File: src/com/music/read/FileLoadUtils.java
package com.music.read;

import java.io.*;

/**
 * Created by xupanpan on 3/24/17.
 */
public class FileLoadUtils {

    public static File getLocalPath() {

        File file = new File(System.getProperty("user.home") + "/.MusicOcean/config/");
        if (!file.exists()) {
            file.mkdirs();
        }
        return file;
    }


    public static String readTextFile(File file) {

        try {
            return readTextFile(new FileInputStream(file));
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
    }


    public static String readTextFile(InputStream inputStream) {
        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();

        byte buf[] = new byte[1024];
        int len;
        try {
            while ((len = inputStream.read(buf)) != -1) {
                outputStream.write(buf, 0, len);
            }
            outputStream.close();
            inputStream.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
        return outputStream.toString();
    }


    public static void saveTextFile(File file, String str) {

        if (file != null && str != null) {
            try {
                FileOutputStream fos = new FileOutputStream(file);
                fos.write(str.getBytes("UTF-8"));
                fos.close();
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
    }

}

File: src/com/music/read/FileLoadUtils.kt
package com.music.read

import java.io.*

/**
 * Created by xupanpan on 3/24/17.
 */
object FileLoadUtils {

    val localPath: File
        get() {

            val file = File(System.getProperty("user.home") + "/.MusicOcean/config/")
            if (!file.exists()) {
                file.mkdirs()
            }
            return file
        }


    fun readTextFile(file: File): String? {

        try {
            return readTextFile(FileInputStream(file))
        } catch (e: Exception) {
            e.printStackTrace()
            return null
        }

    }


    fun readTextFile(inputStream: InputStream): String {
        val outputStream = ByteArrayOutputStream()

        val buf = ByteArray(1024)
        val len: Int
        try {
            while ((len = inputStream.read(buf)) != -1) {
                outputStream.write(buf, 0, len)
            }
            outputStream.close()
            inputStream.close()
        } catch (e: IOException) {
            e.printStackTrace()
        }

        return outputStream.toString()
    }


    fun saveTextFile(file: File?, str: String?) {

        if (file != null && str != null) {
            try {
                val fos = FileOutputStream(file!!)
                fos.write(str!!.toByteArray(charset("UTF-8")))
                fos.close()
            } catch (e: Exception) {
                e.printStackTrace()
            }

        }
    }

}
